src/lib/alloc.c

```diff
@@ -530,10 +538,12 @@ void alloc_trace_buffer_heap(int zone, uint32_t caps, size_t bytes)
 static void *rmalloc_sys_runtime(int zone, int caps, int core, size_t bytes)
 {
 	struct mm_heap *cpu_heap;
-	void *ptr = NULL;
+	void *ptr;
 
 	/* use the heap dedicated for the selected core */
 	cpu_heap = memmap.system_runtime + core;
+	if ((cpu_heap->caps & caps) != caps)
+		return NULL;
```

Comment:
If we can't allocate for system specific request then there should be 

> panic(SOF_IPC_PANIC_MEM);

as far as i understand the system as a whole.

Reply:
thanks, updated